src/sentry/api/endpoints/project_rule_details.py

```diff
@@ -50,9 +52,32 @@ def get(self, request: Request, project, rule) -> Response:
             {method} {path}
 
         """
-        data = serialize(rule, request.user)
 
-        return Response(data)
+        # Serialize Rule object
+        serialized_rule = serialize(
+            rule,
+            request.user,
+        )
+
+        # Prepare Rule Actions that are SentryApp components using the meta fields
+        for action in serialized_rule.data.get("actions", []):
+            if action.get("_sentry_app_installation") and action.get("_sentry_app_component"):
+
+                component = SentryAppInstallation(
+                    **action.get("_sentry_app_installation", {})
```

Comment:
Rehydrating the models from the serialized JSON might cause problem later if the constructor gets a parameter it wasn't prepared for.

Reply:
ehh it seems low risk bc the JSON is from the model originally. Do you have a suggestion on a better way?